src/crawler/Topology.ts

```diff
@@ -1,23 +1,51 @@
 import { getNodeIdFromPeerDescriptor } from '@streamr/dht'
 import { NodeInfo } from '@streamr/trackerless-network'
+import { Multimap } from '@streamr/utils'
 import { DhtAddress, StreamPartID } from 'streamr-client'
 
+interface Node {
+    id: DhtAddress
+    streamPartNeighbors: Multimap<StreamPartID, DhtAddress>
+}
+
 export class Topology {
 
-    private nodeInfos: NodeInfo[]
+    private nodes: Map<DhtAddress, Node> = new Map()
+
+    constructor(infos: NodeInfo[]) {
+        const nodeIds = new Set(...[infos.map((info) => getNodeIdFromPeerDescriptor(info.peerDescriptor))])
+        for (const info of infos) {
+            const streamPartNeighbors: Multimap<StreamPartID, DhtAddress> = new Multimap()
+            for (const streamPartitionInfo of info.streamPartitions) {
+                const neighbors = streamPartitionInfo.deliveryLayerNeighbors
+                    .map((n) => getNodeIdFromPeerDescriptor(n))
+                    .filter((id) => nodeIds.has(id))
+                streamPartNeighbors.addAll(streamPartitionInfo.id as StreamPartID, neighbors)
```

Comment:
I would use `parse` here for safety.

```suggestion
                streamPartNeighbors.addAll(StreamPartIDUtils.parse(streamPartitionInfo.id), neighbors)
```